### PR TITLE
fix: cluster lost should not block close cluster

### PR DESCRIPTION
**Pull request by zhujian7**
**What type of PR is this?**

**What this PR does / why we need it**:

cluster lost should not block close cluster

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @knight42 

**Does this PR introduce a user-facing change?**:

```release-note
Cluster lost does not block close cluster
```

### CODE REVIEW

File: pkg/util/cerr/error.go [hunk 1]
```diff
@@ -181,3 +181,28 @@ func AutoAnalyse(err error) error {
 
 	return err
 }
+
+// TCPConnectionError check whether the err belongs to tcp connection problem.
+func TCPConnectionError(err error) bool {
+	if err == nil {
```
Comment (knight42) [inline pkg/util/cerr/error.go:187]: Is it necessary to check if `err` is nil here?

File: pkg/util/cerr/error.go [hunk 2]
```diff
@@ -181,3 +181,28 @@ func AutoAnalyse(err error) error {
...
+		return false
+	}
+
+	if strings.Contains(err.Error(), "dial tcp") && strings.Contains(err.Error(), "connect: connection refused") {
```
Comment (knight42) [inline pkg/util/cerr/error.go:191]: looks like we could save `err.Error()` to a variable.

**Review by knight42 — APPROVED** [review]

**Comment by zhujian7** [comment]
@knight42 PTAL

File: pkg/util/cerr/error.go [hunk 3]
```diff
@@ -181,3 +181,20 @@ func AutoAnalyse(err error) error {
...
 }
+
+// TCPConnectionError check whether the err belongs to tcp connection problem.
+func TCPConnectionError(err error) bool {
+	e := err.Error()
+	if !strings.Contains(e, "dial tcp"){
+		return false
+	}
+
+	if strings.Contains(e, "connect: connection refused") ||
```
Comment (knight42) [inline pkg/util/cerr/error.go:192]: nit:
```suggestion
	return strings.Contains(e, "connect: connection refused") ||
		strings.Contains(e, "no such host")||
		strings.Contains(e, "i/o timeout")||
		strings.Contains(e, "connect: connection timed out")
```

**Review by knight42 — APPROVED** [review]

File: pkg/util/cerr/error.go (hunk 3 above)
Comment (zhujian7) [reply pkg/util/cerr/error.go]: Fixed

**Pull request merged**